### ISSUE TITLE
Plugin api to extend input types

### ISSUE THREAD

**Issue by olup** [prisma-labs/nexus#371]
Browsing through plugin examples, I was wondering, is there a way to extend input fields as well as output ones ? Something to build custom fields validation, for example ?

My love goes to this library !

**Comment by huv1k** [comment]
I created a plugin for validation of parameters, you can check it out here https://github.com/huv1k/nexus-arguments-validation

**Comment by olup** [comment]
Interesting ! However, if I understand correctly, this allows validation on one mutation / query type. However, it does not let you add validation to a custom input type that could be re-used as argument in any query / mutation type and be validated.

**Comment by Sytten** [comment]
I agree this is missing, I was looking for that. I will add that to my todo list.

**Comment by ahmedosama7450** [comment]
Hey @olup
I recently created a validation plugin, which might be what you're looking for.
[Check it out](https://github.com/ahmedosama7450/nexus-args-validator) and tell me what you think!
